### PR TITLE
Update readme to reflect current style guidelines

### PR DESCRIPTION
**Pull request by h-lame**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,34 +4,419 @@ A web service to map postcodes to administrative boundaries and more.
...
 
-## Development
+## Nomenclature
+
+- **Area**: a set of points that define an area that represents a geopolitical boundary of some sort.  A country, a parliamentary constituency, a national park, a civil parish, etc.
+- **Postcode**: an identifier for a small geographical area usually derived from postal data.  Postcodes don't have boundaries, instead they have a lat/lng point which is the centroid of their area.  The **Area**s for that **Postcode** are looked up by doing a geo search for those that contain the point.
+- **Generation**: because areas change over time we add **Area**s to generations.  If an area is in one generation but not another it is likely because legislation was enacted between those generations that meant it ceased to exist (often it is replaced by another similar but not exactly the same Area).  Unless specified in the request mapit only searches in the current generation.
+- **Type**: to identify what an area is it has a type.  For example European parliamentary constituencies are type 'EUR' comapred to Westminister parliamentary constituencies of type 'WMC'.
```
Comment (sihugh) [inline README.md]: Typo - comapred -> compared

File: README.md [hunk 2]
```diff
@@ -4,34 +4,419 @@ A web service to map postcodes to administrative boundaries and more.
...
+- **CodeType**: Areas have codes to allow them to be identified, issued by the ONS.  Prior to 2011 these were SNAC codes (stored as type 'ons') (see: http://www.ons.gov.uk/ons/guide-method/geography/products/names--codes-and-look-ups/standard-names-and-codes/index.html), since 2011 these are GSS codes (stored as type 'gss') (see: http://neighbourhood.statistics.gov.uk/dissemination/Info.do?m=0&s=1460381709047&enc=1&page=nessgeography/new-geography-codes-and-naming-policy-from-1-january-2011.htm&nsjs=true&nsck=false&nssvg=false&nswid=1212)
+- **ONSPD**: The Office for National Statistics Postcode Database - one of the open data sources we use to import data. We get UK postcodes from this.
+- **Boundary Line**: The Ordnance Survey boundary line data - one of the open data sources we use to import data.  We get GB Areas from this.
+- **OSNI**: The Ordnance Survey of Northern Ireland - one of the providers of the open data srouces we use to import data.  We get NI Areas from them.
```
Comment (sihugh) [inline README.md]: Typo - srouces -> sources

**Comment by sihugh** [comment]
Just a couple of typos, but other than that it looks great.

File: README.md (hunk 1 above)
Comment (h-lame) [inline README.md]: Thanks.

File: README.md (hunk 2 above)
Comment (h-lame) [inline README.md]: Thanks!

**Pull request merged**